openstack/dms/v1/products/requests.go

```diff
@@ -6,6 +6,8 @@ import (
 
 // Get products
 func Get(client *golangsdk.ServiceClient, engine string) (r GetResult) {
-	_, r.Err = client.Get(getURL(client, engine), &r.Body, nil)
+	url := getURL(client)
+	url = url + "?engine=" + engine
```

Comment:
Please consider the case where the `engine` is empty

Reply:
The logic has not change, and engine is a required parameter

Reply:
This assumption is unreliable, we cannot assume that the caller always uses a non-empty value.

Reply:
OK